### PR TITLE
cassandra-cpp-driver: init at 2.16.2

### PR DESCRIPTION
**Pull request by npatsakula**
###### Description of changes

Initial implementation of the DataStax CPP Cassandra driver derivation.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/cassandra-cpp/default.nix
Comment (symphorien) [inline pkgs/development/libraries/cassandra-cpp/default.nix:1]: you should not take `pkgs` as argument but all the members you need: `stdenv, fetchgit, ...`

File: pkgs/development/libraries/cassandra-cpp/default.nix
Comment (symphorien) [inline pkgs/development/libraries/cassandra-cpp/default.nix:14]: zlib libuv and openssl are presumably buildInputs. You can set `strictDeps = true` to check.

File: pkgs/development/libraries/cassandra-cpp/default.nix
Comment (symphorien) [inline pkgs/development/libraries/cassandra-cpp/default.nix:19]: I leave it up to you, but this looks over-engineered.

File: pkgs/development/libraries/cassandra-cpp/default.nix
Comment (symphorien) [inline pkgs/development/libraries/cassandra-cpp/default.nix:22]: I think this is the default with cmake

File: pkgs/development/libraries/cassandra-cpp/default.nix
Comment (symphorien) [inline pkgs/development/libraries/cassandra-cpp/default.nix:23]: can you add meta.homepage ?

File: pkgs/development/libraries/cassandra-cpp/default.nix
Comment (symphorien) [inline pkgs/development/libraries/cassandra-cpp/default.nix:5]: Any reason not to use fetchFromGitHub ? (it is more bandwidth efficient)

**Review by symphorien — COMMENTED** [review]
Thanks for your PR. I left a few comments.

File: pkgs/development/libraries/cassandra-cpp/default.nix
Comment (npatsakula) [reply pkgs/development/libraries/cassandra-cpp/default.nix]: No one. Fixed!

File: pkgs/development/libraries/cassandra-cpp/default.nix
Comment (npatsakula) [reply pkgs/development/libraries/cassandra-cpp/default.nix]: Done.

File: pkgs/development/libraries/cassandra-cpp/default.nix
Comment (npatsakula) [reply pkgs/development/libraries/cassandra-cpp/default.nix]: Fixed.

File: pkgs/development/libraries/cassandra-cpp/default.nix
Comment (npatsakula) [reply pkgs/development/libraries/cassandra-cpp/default.nix]: Roger that. Fixed.

**Review by symphorien — APPROVED** [review]
Looks good to me.

Two minor comments:
- other distributions call this package cassandra-cpp-driver https://repology.org/project/cassandra-cpp-driver/information would you mind renaming this one to match?
- according to the manual, this PR should have two commits
  * `maintainers: add npatsakula`
  * `cassandra_cpp: init at 2.16.2`

I'll leave this open for a few days anyway in case someone else has comments.

**Comment by npatsakula** [comment]
@symphorien, I've fixed all comments.

**Review by viraptor — COMMENTED** [review]
Result of `nixpkgs-review pr 176535` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cassandra-cpp-driver</li>
  </ul>
</details>

**Review by SuperSandro2000 — COMMENTED** [review]
At least my comment in maintainer-list.nix and all-packages.nix should have blocked the merging of this PR. The three comments in the package would have been nice to have.

**Pull request merged**
